QRCoder/SvgQRCode.cs

```diff
@@ -214,54 +212,229 @@ public string GetGraphic(Size viewBox, string darkColorHex, string lightColorHex
-        var imgHeight = logo.GetIconSizePercent() / 100d * viewBox.Height;
-        var imgPosX = viewBox.Width / 2d - imgWidth / 2d;
-        var imgPosY = viewBox.Height / 2d - imgHeight / 2d;
-        return new ImageAttributes()
+            // Build path using RLE to combine rectangles
+            for (int y = 0; y < drawableModulesCount; y++)
+            {
+                int x = 0;
+
+                while (x < drawableModulesCount)
+                {
+                    if (!ShouldDrawModule(x, y))
+                    {
+                        x++;
+                        continue;
+                    }
+
+                    // Found a module to draw - find the run length
+                    int startX = x;
+                    x++;
+
+                    while (x < drawableModulesCount && ShouldDrawModule(x, y))
+                    {
+                        x++;
+                    }
+
+                    int width = x - startX;
+                    if (width > 0)
+                    {
+                        // Absolute move to start of rectangle
+                        svgFile.Append('M');
+                        Append(startX);
+                        svgFile.Append(' ');
+                        Append(y);
+
+                        // Draw rectangle using relative movements (width, height of 1)
+                        svgFile.Append('h');
+                        Append(width);
+                        svgFile.Append("v1h-");
+                        Append(width);
+                        svgFile.Append('z');
+                    }
+                }
+            }
+
+            svgFile.AppendLine(@"""/>");
+
+            // Local function to determine if a module should be drawn
+            bool ShouldDrawModule(int x, int y)
+            {
+                bool isDarkModule = QrCodeData.ModuleMatrix[y + offset][x + offset];
+                bool isBlockedByLogo = logo != null && logo.FillLogoBackground() &&
+                    IsBlockedByLogo(x, y, logoAttr!.Value);
+
+                // For dark modules: draw only dark modules not blocked by logo
+                // For light modules: draw light modules not blocked by logo, or anywhere blocked by logo (regardless of module color)
+                return drawDarkModules
+                    ? (isDarkModule && !isBlockedByLogo)
+                    : (!isDarkModule || isBlockedByLogo);
+            }
+        }
+
+        // Local function to append integers efficiently
+        void Append(int num)
         {
-            Width = imgWidth,
-            Height = imgHeight,
-            X = imgPosX,
-            Y = imgPosY
-        };
+#if HAS_SPAN
+            Span<char> buffer = stackalloc char[16];
+            if (num.TryFormat(buffer, out int charsWritten, default, CultureInfo.InvariantCulture))
+            {
+                svgFile.Append(buffer.Slice(0, charsWritten));
+                return;
+            }
+#endif
```

Comment:
```suggestion
#if NET6_0_OR_GREATER
            svgFile.Append(CultureInfo.InvariantCulture, $"{num}");
#endif
```
Should also (w/o measuring) be a little bit faster, as it's formatted to the SB's buffer directly, and not via the span detour.

For < .NET 6 where span is available I wouldn't bother and don't have a separate #ifdef for this.

Reply:
I pushed this code into StringExtensions, with a fast path for .NET 6+, a fallback for HAS_SPAN, and fallback for anything else.  Your argument is completely valid; we really don't need to optimize for older targets....but why not - so long as it doesn't impede reading the code?  I think what I'd like to do is for v2 create an `Append(IFormatProvider, FormattableString)` method for pre-.NET6 environments and combine a bunch of these multi-line Append commands into a interpolated string for better readability and more concise code.

Reply:
I'd also like to consider a dependency on System.Memory for .NET Standard 2.0 support in v2 so that Span support is available throughout the codebase.  We can review a PR and see if it makes sense or not.  It may not help much if the framework (like int.Parse) does not support Span.

Reply:
> if the framework (like int.Parse) does not support Span.

I think that could be polyfilled with C#'s extension members. So a new compiler creaets code that runs on old targets.
Thus I also think that System.Memory makes the code base cleaner.